### PR TITLE
Make Reduced Network from partition

### PR DESCRIPTION
**Pull request by sfwolfphys**
I've added the function `make_reduced_from_partition` to the repo, and I have created all the appropriate docs, updated the NEWS, and added myself as a package co-author.  I don't know if I should be coauthor or contributor or whatnot, so feel free to change that.  [This paper](https://journal.r-project.org/archive/2012-1/RJournal_2012-1_Hornik~et~al.pdf) has some potentially useful definitions on that front.  This doesn't seem to have changed the package citation.  Indeed, I'll let that be a "before you push to CRAN change.  I see a file `inst\CITATION`, but I don't know if it gets automagically generated based on the description when it moves to CRAN, or if it gets hand edited.  I don't want to make it break by hand editing ;)

### CODE REVIEW

**Comment by sfwolfphys** [comment]
Also, this links my [proof of concept repo](https://github.com/sfwolfphys/mrfp/) which has a badge on it to my current Travis build status.  (I will link this here too...)

[Travis build status](https://travis-ci.com/sfwolfphys/concorR)

File: man/makereducedfrompartition.Rd [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+reduced adjacency matrix and the corresponding cutoff density/degree. The 
+\pkg{sna} package must be installed, but does not need to be attached.
+}
+\usage{
+make_reduced_from_partition(adj_mat, partition, stat = 'density')
+}
+\arguments{
+  \item{adj_mat}{A adjacency matrix}
+  \item{partition}{A vector that gives the desired partitioning of the nodes. }
+  \item{stat}{The statistic that will be used to determine if edges should be included in a reduced network.  Default value is \code{density}}
```
Comment (atraxler) [inline man/makereducedfrompartition.Rd:15]: Note that the other implemented option is `degree`?

File: man/makereducedfrompartition.Rd [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+\item{reduced_mat}{A reduced matrix}
+\item{dens}{The cut-off density used (equal to the edge density of \code{adj_mat}).  Only for \code{stat="density"}}
+\item{deg}{The cut-off normalized outdegree used (equal to the mean outdegree of \code{adj_mat}).  Only for \code{stat="degree"}.}
+}
+\references{
+S. Wasserman and K. Faust, \emph{Social Network Analysis: Methods and Applications} (Cambridge University Press, 1994).
+}
+
+\seealso{
+  \code{\link{concor}}, \code{\link{make_blk}}
```
Comment (atraxler) [inline man/makereducedfrompartition.Rd:31]: Can probably nix `concor` and maybe `make_blk`, but add `make_reduced`?

File: NEWS.md [hunk 3]
```diff
@@ -6,6 +6,8 @@ missing.
 they had to be added.
 * Added a `stat` option to `make_reduced()`, so it can use density to threshold 
 reduced network links (the original behavior) or average degree.
+* Updated `plot_reduced()` color pallette so that it auto-generates the number of needed colors. Using viridis now instead of rainbow.
```
[the remark this answers is not in the document]
Comment (sfwolfphys) [reply NEWS.md:9]: Fixed in next commit

File: man/makereducedfrompartition.Rd [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+g1 <- matrix(c(0, 0, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 1, 0, 1, 0), ncol = 4)
+rownames(g1) <- c("a", "b", "c", "d")
+colnames(g1) <- c("a", "b", "c", "d")
+
+g2 <- matrix(c(0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1, 1, 0, 0), ncol = 4)
+rownames(g2) <- c("a", "b", "c", "d")
+colnames(g2) <- c("a", "b", "c", "d")
+
+make_reduced(list(g1, g2), nsplit = 1)
+make_reduced(list(g1, g2), nsplit = 1, stat="degree")
```
[the remark this answers is not in the document]
Comment (sfwolfphys) [reply man/makereducedfrompartition.Rd:43]: Using example from my proof of concept repo.

File: R/CONCOR_blockmodeling.R [hunk 5]
```diff
@@ -153,8 +155,106 @@ make_reduced_igraph <- function(reduced_mat) {
...
+  d <- a / m
+  return(d)
+}    
+
+#' @export
+make_reduced_from_partition <- function(adj_mat, partition, stat='density') {
+  if(stat=='density'){  
+      dens <- .edge_dens(adj_mat)
+
+      nb = max(partition)
```
[the remark this answers is not in the document]
Comment (sfwolfphys) [reply R/CONCOR_blockmodeling.R:177]: Agreed.  There is new code in next push.

File: man/makereducedfrompartition.Rd [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+Takes an adjacency matrix and a partition of the nodes, and returns a
+reduced adjacency matrix and the corresponding cutoff density/degree. The 
+\pkg{sna} package must be installed, but does not need to be attached.
+}
+\usage{
+make_reduced_from_partition(adj_mat, partition, stat = 'density')
+}
+\arguments{
+  \item{adj_mat}{A adjacency matrix}
+  \item{partition}{A vector that gives the desired partitioning of the nodes. }
```
[the remark this answers is not in the document]
Comment (sfwolfphys) [reply man/makereducedfrompartition.Rd:14]: agree

File: man/makereducedfrompartition.Rd (hunk 1 above)
Comment (sfwolfphys) [reply man/makereducedfrompartition.Rd:15]: agree

File: man/makereducedfrompartition.Rd (hunk 2 above)
Comment (sfwolfphys) [reply man/makereducedfrompartition.Rd:31]: done

**Review by atraxler — APPROVED** [review]

**Pull request merged**